### PR TITLE
feat[sign in with social accounts]                                   …

### PR DESCRIPTION
**Pull request by FabCode67**
### description of task

- allow user to signup with google account
- write its swagger documentation
- configure google console settings

- this branch contains middleware file for setting authentications to user there is also routes for directing a user to preferred pages 

### how to run it

- `git clone` from repo
- `git checkout ft-signin-with-social-accounts`
- set .env as shown in `.envexample`
- run `npm i `if you don't have configured docker in your machine run `npm run dev`
- copy this URL in your browser: http://localhost:{port}/auth/google chose your email and test it

## feat[finish sign in with social accounts]

### CODE REVIEW

File: src/controllers/Auth/google_oauth.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import db from '../../models';
+
+export const loginWithGoogle = async (req, res) => {
+    try {
+      const { user, token } = req.user;
+      const existingUser = await db.user.findOne({ where: { email: user.email } });
+      if(!existingUser){
+        res.status(302).json({status:"fail", message: req.t('user_email_not_found')})
```
Comment (Aimeana100) [inline src/controllers/Auth/google_oauth.js:8]: Thank you @FabCode67 for the great work that you had really done!! I am wondering if 302 is the right response status for the case a user tries to sign in with account unexist.

**Review by Evaristekanova — APPROVED** [review]
This PR is working correct

**Review by Aimeana100 — APPROVED** [review]
Codes runs smoothly as expected 👍

**Review by Kevin-ishimwe — APPROVED** [review]

**Review by BenoitIT — APPROVED** [review]

File: src/controllers/Auth/google_oauth.js (hunk 1 above)
Comment (FabCode67) [reply src/controllers/Auth/google_oauth.js:8]: > Thank you @FabCode67 for the great work that you had really done!! I am wondering if 302 is the right response status for the case a user tries to sign in with account unexist.

yes @Aimeana100 but check this info: _the HyperText Transfer Protocol (HTTP) 302 Found redirect status response code indicates that the resource requested has been temporarily moved to the URL given by the Location header._
_Since the 302 Found indicates something has gone wrong within your application's server, we can disregard the client-side._

**Review by mansurissa — APPROVED** [review]
Good job

**Pull request merged**